### PR TITLE
Refactor l10n code. Remove use of react-intl.

### PR DESCRIPTION
**Pull request by adomasven**
Strings from .dtd files are now accessible from Zotero.getMessage as the eventual move to Fluent would remove their distinction.

Throws if there are duplicate string ids between .dtd and .properties files

### CODE REVIEW

**Comment by adomasven** [comment]
I kept the `react-intl` package there for now as plugin devs that used VirtualizedTable would have needed to use the `IntlProvider` class from it. We should remove it in the future.

File: chrome/content/zotero/xpcom/intl.js [hunk 1]
```diff
@@ -82,6 +82,13 @@ Zotero.Intl = new function () {
 			let regexp = /<!ENTITY ([^\s]+)\s+"([^"]+)/g;
 			let regexpResult;
 			while (regexpResult = regexp.exec(localeXML)) {
+				let bundleString;
+				try {
+					bundleString = bundle.GetStringFromName(regexpResult[1]);
+				} catch (e) {}
+				if (bundleString) {
+					throw new Error(`Found a duplicate string ${regexpResult[1]} for both bundle and dtd strings`)
+				}
```
Comment (dstillman) [inline chrome/content/zotero/xpcom/intl.js:91]: We should just add this to a test. It's only ~4 milliseconds for me, but no reason to do it on every startup, and milliseconds add up eventually.

File: chrome/content/zotero/xpcom/intl.js [hunk 2]
```diff
@@ -105,6 +112,9 @@ Zotero.Intl = new function () {
 				}
 				l10n = bundle.formatStringFromName(name, params, params.length);
 			}
+			else if (Zotero.Intl.strings[name]) {
+				return Zotero.Intl.strings[name];
+			}
```
Comment (dstillman) [inline chrome/content/zotero/xpcom/intl.js:117]: `this`

File: chrome/content/zotero/xpcom/intl.js (hunk 1 above)
Comment (adomasven) [reply chrome/content/zotero/xpcom/intl.js:91]: We're initializing the strings here anyway so I didn't think it egregious. If it's a test you'll end up not noticing you've added a duplicate until pushing to github and the test suite running, which is like a whole lot more compute, if we're concerned about that.

Comment (dstillman) [reply chrome/content/zotero/xpcom/intl.js:91]: It's totally fine, but there's a (very small) measurable difference for just this block, so we might as well avoid it.

Given that there was a single duplicate in 15 years of strings, I think it's fine for it not to be caught until CI (and that's only necessary if you mess it up in the direction where you wouldn't immediately notice it in the UI you're working on).

**Comment by adomasven** [comment]
Updated.

**Pull request merged**